File: src/main/java/io/tripled/social/client/application/DefaultReadWallUseCase.java
package io.tripled.social.client.application;

import io.tripled.social.client.domain.SocialNetwork;
import io.tripled.social.client.domain.SocialNetworkRepository;
import io.tripled.social.client.domain.UserName;

public class DefaultReadWallUseCase implements ReadWallUseCase {

  private SocialNetworkRepository socialNetworkRepository;

  public DefaultReadWallUseCase(SocialNetworkRepository socialNetworkRepository) {
    this.socialNetworkRepository = socialNetworkRepository;
  }

  @Override
  public String readWall(String userName) {
    SocialNetwork socialNetwork = socialNetworkRepository.get();

    return socialNetwork.readWall(new UserName(userName));
  }
}

File: src/main/kotlin/io/tripled/social/client/application/DefaultReadWallUseCase.kt
package io.tripled.social.client.application

import io.tripled.social.client.domain.SocialNetwork
import io.tripled.social.client.domain.SocialNetworkRepository
import io.tripled.social.client.domain.UserName

class DefaultReadWallUseCase(socialNetworkRepository: SocialNetworkRepository) : ReadWallUseCase {
    private val socialNetworkRepository: SocialNetworkRepository

    init {
        this.socialNetworkRepository = socialNetworkRepository
    }

    override fun readWall(userName: String): String {
        val socialNetwork: SocialNetwork = socialNetworkRepository.get()

        return socialNetwork.readWall(UserName(userName))
    }
}
